File: Spring boot/spring-boot-ecommerce/src/main/java/com/project/ecommerce/entity/Product.java
package com.project.ecommerce.entity;

import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import javax.persistence.*;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "products")
@Getter
@Setter
public class Product {

    @Id
    private String id;

    private String name;

    @Column
    private String description;

    private String sku;

    private String unitPrice;

    private String imageUrl;

    private boolean active;

    private int unitsInStock;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id")
    private Category category;

    @CreationTimestamp
    private LocalDateTime createdDate;

    @UpdateTimestamp
    private LocalDateTime lastUpdatedDate;

    private String fileName;

    public Product(){
        this.id= UUID.randomUUID().toString();
    }
}

File: angular/angular-ecommerce/src/app/common/product.ts
export class Product {
  constructor(
    public id:string,
    public name: string,
    public descrption: string,
    public sku: string,
    public unitPrice: string,
    public imageUrl: string,
    public active: boolean,
    public unitsInStock: number,
    public createdDate:Date,
    public lastUpdatedDate:Date) {}
}
